sdk/java-sdk/src/main/java/kalix/javasdk/valueentity/ValueEntity.java

```diff
@@ -171,7 +173,19 @@ protected final Effect.Builder<S> effects() {
        * @param <T> The type of the message that must be returned by this call.
        * @return An error reply.
        */
-      <T> Effect<T> error(String description, Status.Code statusCode);
+      <T> Effect<T> error(String description, Status.Code grpcErrorCode);
+
+      /**
+       * Create an error reply with a custom status code.
+       * This status code will be translated to an HTTP or gRPC code
+       * depending on the type of service being exposed.
+       *
+       * @param description The description of the error.
+       * @param errorCode A custom Kalix status code to represent the error.
+       * @return An error reply.
+       * @param <T> The type of the message that must be returned by this call.
+       */
+      <T> Effect<T> error(String description, StatusCode.ErrorCode httpErrorCode);
```

Comment:
The javadoc tries to be more generic by saying that it will be converted to Http or gRPC, but the param name is `httpErrorCode`. And the names are aligned with http error codes. 

I think we can just refer to it as Http Error code instead. 

Eventually, a gRPC user will use this method and handle exceptions of type `StatusRuntimeException`. But that's fine. Such a user will be already passing `ErrorCode.BAD_REQUEST` on one side and handling `io.grpc.Status.Code.INVALID_ARGUMENT` on the other side.